### DISCUSSION TITLE
Unique Node Names

### DISCUSSION THREAD

**Discussion by tpfau** [newcat/baklavajs#279] [Q&A] [answered]
I have the following situation:
I would like to have a graph with unique Node Titles (i.e. no two nodes should have the exact same title). I know that there is already a node ID which is unique, but for usability I would need a human readable unique identifier as I would like users to be able to reference these nodes by their title, and thus they would need to be unique. 
From what I have seen, there is (currently) no option to allow preventing a rename action on a vue node, and since this is a direct modification of the underlying node it also seems as the only way to prevent this would be to react to the change and undo it. somehow this feels wrong, and I'm wondering, whether there is a better way. 
I could of course try to build my own Node visualisation class, but I'm wondering, whether there is a simpler way.

**Comment by newcat** [comment, marked as answer]
Good point, I didn't have such a use-case in mind so I didn't create `beforeRename`/`rename` events. But I'll add them in the next version
